test_cases/stdlib/collections/check_counter.py

```diff
@@ -0,0 +1,37 @@
+from __future__ import annotations
+
+from collections import Counter
+from typing_extensions import assert_type
+
+
+class Foo: ...
+
+
+# Test the constructor
+# assert_type(Counter(), Counter[Never, int])  # pyright derives "Unknown" instead of "Never"
```

Comment:
For what it's worth I think that's better than `Never`. We could test the value type with something like:
```
for value in Counter().values:
    assert_type(value, int)
```

Reply:
Unfortunately, this makes pyright unhappy as well.

Reply:
`assert_type(Counter(), "Counter[Any, int]")` works.